src/python/pants/backend/jvm/tasks/coverage/jacoco.py

```diff
@@ -152,7 +164,16 @@ def _get_target_classpaths(self):
     for target in self._coverage_targets:
       paths = runtime_classpath.get_for_target(target)
       for (name, path) in paths:
-        target_paths.append(path)
+        if len(self._target_filters) == 0:
+          target_paths.append(path)
+        else:
+          for filter in self._target_filters:
+            if re.search(filter, path) is not None:
```

Comment:
This is matching the regex against the classpath entry filename... if you intended to match against the string form of the target address (which is what the option help suggests), you'd want to match against `target.address.spec`.

Reply:
If you are comparing just the target specs, you could check when `self._coverage_targets` is populated instead of here.